### PR TITLE
Fixed sonar issues in identity-outbound-auth-tiqr

### PR DESCRIPTION
**Pull request by supunharsha**

### CODE REVIEW

File: component/authentication-endpoint/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/tiqr/QRCode.java [hunk 1]
```diff
@@ -203,7 +203,9 @@ private String sendRESTCall(String url, String urlParameters, String formParamet
             }
             return TiqrConstants.FAILED + e.getMessage();
         } finally {
-            connection.disconnect();
+            if(connection != null){
```
Comment (omindu) [inline component/authentication-endpoint/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/tiqr/QRCode.java]: Formatting issue `if(connection != null) {`

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/tiqr/TiqrAuthenticator.java [hunk 2]
```diff
@@ -373,7 +373,10 @@ private String sendRESTCall(String url, String urlParameters, String formParamet
             }
             return TiqrConstants.FAILED + e.getMessage();
         } finally {
-            connection.disconnect();
+            if(connection != null){
```
Comment (omindu) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/tiqr/TiqrAuthenticator.java]: Formatting issue `if(connection != null) {`

**Review by omindu — CHANGES_REQUESTED** [review]

File: component/authentication-endpoint/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/tiqr/QRCode.java [hunk 3]
```diff
@@ -203,7 +203,9 @@ private String sendRESTCall(String url, String urlParameters, String formParamet
             }
             return TiqrConstants.FAILED + e.getMessage();
         } finally {
-            connection.disconnect();
+            if (connection != null) {
```
Comment (madumalt) [inline component/authentication-endpoint/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/tiqr/QRCode.java:206]: can't we go with, "try with" here?

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/tiqr/TiqrAuthenticator.java [hunk 4]
```diff
@@ -373,7 +373,10 @@ private String sendRESTCall(String url, String urlParameters, String formParamet
             }
             return TiqrConstants.FAILED + e.getMessage();
         } finally {
-            connection.disconnect();
+            if (connection != null) {
```
Comment (madumalt) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/tiqr/TiqrAuthenticator.java:376]: can't we go with, "try with" here?

File: component/authentication-endpoint/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/tiqr/QRCode.java (hunk 3 above)
Comment (supunharsha) [reply component/authentication-endpoint/src/main/java/org/wso2/carbon/identity/application/authentication/endpoint/tiqr/QRCode.java:206]: HttpURLConnection is not implements from AutoClosable interface, so we can't use it with try with resource.

File: component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/tiqr/TiqrAuthenticator.java (hunk 4 above)
Comment (supunharsha) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/authenticator/tiqr/TiqrAuthenticator.java:376]: HttpURLConnection is not implements from AutoClosable interface, so we can't use it with try with resource.

**Review by rushmin — APPROVED** [review]

**Review by omindu — APPROVED** [review]

**Pull request merged**